app/controllers/chargeback_controller.rb

```diff
@@ -255,8 +255,19 @@ def cb_rate_form_field_changed
     cb_rate_get_form_vars
     render :update do |page|                    # Use JS to update the display
       changed = (@edit[:new] != @edit[:current])
+      # Update the new column with the code of the currency selected by the user
+      @edit[:new][:details].each_with_index do |_detail, i|
+        next unless @edit[:new][:details][i][:currency] != @edit[:current][:details][i][:currency]
+        @edit[:current][:details][i][:currency] = @edit[:new][:details][i][:currency]
+        params[:id_column] = i
+        id_currency = @edit[:current][:details][i][:currency]
+        currency = ChargebackRateDetailCurrency.find_by(:id => id_currency)
+        params[:code_currency] = currency.code
+        page.replace("column_currency_#{i}", :partial => "cb_new_currency_column")
```

Comment:
Perhaps this block should begin with creating two local variables so we don't have to have such deeply nested hash access.  I had to read the hash accces [][] multiple times to see which ones were the same and which were different.

I think something like the below makes it clearer what is happening:

```ruby
new_rate_details = @edit[:new][:details][i]
old_rate_details = @edit[:current][:details][i]
next unless new_rate_details[:currency] != current_rate_details[:currency]

current_rate_details[:currency] = new_rate_details[:currency]
currency = ChargebackRateDetailCurrency.find_by(:id => current_rate_details[:currency])
params[:code_currency] = currency.code
params[:id_column] = i
page.replace("column_currency_#{i}", :partial => "cb_new_currency_column")
```